src/plugins/mf_priority.cpp

```diff
@@ -950,6 +954,49 @@ static int validate_cb (flux_plugin_t *p,
                                            "user has max active jobs");
     }
 
+    if (jobspec == NULL) {
+        flux_jobtap_reject_job (p,
+                                args,
+                                "job.validate: failed to unpack jobspec");
+    } else {
+        // if a queue was not unpacked, just set it to ""
+        queue_str = queue ? queue : "";
+        // count resources requested for the job
+        if (job.count_resources (jobspec) < 0) {
+            return flux_jobtap_reject_job (p,
+                                           args,
+                                           "job.validate: unable to count " \
+                                           "resources for job");
+        }
+        if (job.ncores > 0 && job.nnodes == 0)
+            // the job specified cores but no nodes, so we need to set
+            // nnodes == 1 here
+            job.nnodes = 1;
+        // look up queue in queues map to see if it has a defined
+        // max_nodes_per_association limit
+        auto it = queues.find(queue_str);
+        if (it != queues.end ()) {
+            if (job.nnodes > queues[queue_str].max_nodes_per_assoc) {
+            // the job size is greater than the max nodes per-association limit
+            // configured for this queue; reject the job
+            return flux_jobtap_reject_job (p,
+                                           args,
+                                           "job.validate: job size is greater "
+                                           "than max resources limit "
+                                           "configured for queue");
+            }
```

Comment:
Is it possible (or desirable I guess) to report the limit and the current job size in this error? e.g.

 nodes requested (24) greater than maximum (16) configured for queue

Also, I'd drop the `job.validate` prefix from this error since it will be returned to the user (and the error will already say something like "submission failed")